### ISSUE TITLE
wrong answer of argmax for big CPU tensors with dim=2

### ISSUE THREAD

**Issue by Heerozh** [pytorch/pytorch#38922]
## 🐛 Bug

Reproducible on both 1.5.0 and 1.6.0dev20200522

```
import torch
x = torch.full((40, 900, 1), 0, dtype=torch.float32)
x.argmax(dim=2)
```

tensor([[    0,     0,     0,  ...,     0,     0,     0],
        [    0,     0,     0,  ...,     0,     0,     0],
        [    0,     0,     0,  ...,     0,     0,     0],
        ...,
        [32400, 32400, 32400,  ..., 32400, 32400, 32400],
        [32400, 32400, 32400,  ..., 32400, 32400, 32400],
        [32400, 32400, 32400,  ..., 32400, 32400, 32400]])

## Expected

```
import torch
x = torch.full((40, 900, 1), 0, dtype=torch.float32)
x.numpy().argmax(axis=2)
```
array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]], dtype=int64)

```
x = torch.full((36, 900, 1), 0, dtype=torch.float32)
x.argmax(dim=2)
```
tensor([[0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0],
        ...,
        [0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0]])

## Environment

PyTorch version: 1.6.0.dev20200522
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce RTX 2080 Ti
Nvidia driver version: 445.87
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.6.0.dev20200522
[pip] torchvision==0.7.0.dev20200522
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               10.2.89              h74a9793_1
[conda] mkl                       2020.1                      216
[conda] mkl-service               2.3.0            py37hb782905_0
[conda] mkl_fft                   1.0.15           py37h14836fe_0
[conda] mkl_random                1.1.0            py37h675688f_0
[conda] numpy                     1.18.1           py37h93ca92e_0
[conda] numpy-base                1.18.1           py37hc3f5095_1
[conda] pytorch                   1.6.0.dev20200522 py3.7_cuda102_cudnn7_0    pytorch-nightly
[conda] torchvision               0.7.0.dev20200522      py37_cu102    pytorch-nightly

cc @ezyang @gchanan @zou3519 @VitalyFedyunin

**Comment by SsnL** [comment]
It is not "wrong". There is currently no guarantee which indices to return upon ties for argmax/min. See also https://github.com/pytorch/pytorch/issues/10370

**Comment by Heerozh** [comment]
> It is not "wrong". There is currently no guarantee which indices to return upon ties for argmax/min. See also #10370

32400 is out of bound, i think it should be the wrong value?

**Comment by SsnL** [comment]
Ohh I see. My bad. I didn't read the input shape closely. You are right that this is a bug.
